Duplicati/UnitTest/UtilityTests.cs

```diff
@@ -125,6 +125,164 @@ public static void ReadLimitLengthStreamWithGrowingFile()
             Assert.IsTrue(limitStreamLength < nextGrowingStreamLength);
         }
 
+        [Test]
+        [Category("Utility")]
+        public static void ReadLimitLengthStream()
+        {
+            Action<IEnumerable<int>, IEnumerable<byte>> assertArray = (expected, actual) =>
+            {
+                List<int> expectedList = expected.ToList();
+                List<byte> actualList = actual.ToList();
+                Assert.AreEqual(expectedList.Count, actualList.Count, "Count");
+                for (int i = 0; i < expectedList.Count; i++)
+                {
+                    Assert.AreEqual((byte)expectedList[i], actualList[i], "Index {0}", i);
+                }
+            };
+
+            byte[] readBuffer = new byte[10];
+            Action<Stream, int, int> testSeek = (stream, position, readByte) =>
+            {
+                stream.Position = position;
+                Assert.AreEqual(position, stream.Position);
+                Assert.AreEqual(readByte, stream.ReadByte());
+
+                Assert.AreEqual(position, stream.Seek(position, SeekOrigin.Begin));
+                Assert.AreEqual(position, stream.Position);
+                Assert.AreEqual(readByte >= 0 ? 1 : 0, stream.Read(readBuffer, 0, 1), "Read count");
+                if (readByte >= 0)
+                {
+                    Assert.AreEqual(readBuffer[0], readByte);

```

Comment:
Nice job with the tests.  Would it be a slight improvement in coverage to provide an offset of `1` to `stream.Read` on line 152, and check `readBuffer[1]` here to verify that the offset is used correctly?

Reply:
That's not a bad idea. I've also set it to clear the array first and make sure the indexes before and after are clear (and make sure nothing is read when Read returns 0.)